front/src/pages/teachers/teachers-information/index.js

```diff
@@ -0,0 +1,263 @@
+    this.renderChangeNameOrActualName = this.renderChangeNameOrActualName.bind(this);
+    this.renderChangeEmailOrActualEmail = this.renderChangeEmailOrActualEmail.bind(this);
+    this.changeEmail = this.changeEmail.bind(this);
+    this.cancelEmailChanging = this.cancelEmailChanging.bind(this);
+    this.handleEmailChange = this.handleEmailChange.bind(this);
+    this.showModalDelete = this.showModalDelete.bind(this);
+    this.hideModalDelete = this.hideModalDelete.bind(this);
+    this.handleDeleteTeacher = this.handleDeleteTeacher.bind(this);
+  }
+
+  changeName(event) {
+    const id = +event.target.dataset.id;
+    const { changeName } = this.props;
+    changeName(id);
+  }
+
+  cancelNameChanging(event) {
+    const id = +event.target.dataset.id;
+    const { cancelNameChanging } = this.props;
+    cancelNameChanging(id);
+  }
+
+  handleNameChange(data) {
+    const { updateName, updatingId } = this.props;
+    const arr = data.nameSurname.split(' ');
+    const [firstName, lastName] = arr;
+    data = {
+      firstName,
+      lastName,
+      id: updatingId
+    };
+
+    updateName(data);
+  }
+
+  rememberId(event) {
+    const id = +event.target.dataset.id;
+    this.props.rememberUpdatingId(id)
+  }
+
+  renderChangeNameOrActualName(id, changeNameIds) {
+    if (changeNameIds.includes(id)) {
+      return (
+        <>
+          <Form {...layout} onFinish={this.handleNameChange}>
+            <Item
+              name="nameSurname"
+              align="center"
+              rules={[
+                { required: true, message: 'Please input new name and surname in English!' },
+                { validator: validateNameSurname },
+              ]}
+            >
+              <Input placeholder="Name in English" />
+            </Item>
+            <Form.Item align="center">
+              <SaveButton data-id={id} onClick={this.rememberId} htmlType="submit">
+                <SaveOutlined />
+              </SaveButton>
+            </Form.Item>
+          </Form>
+          <CancelButton data-id={id} onClick={this.cancelNameChanging}>
+            <RollbackOutlined />
+          </CancelButton>
+        </>
+      );
+    } else {
+      const { firstName, lastName } = this.props;
+      return (
+        <>
+          <NameSurname>{`${firstName} ${lastName}`}</NameSurname>
+          <EditButton data-id={id} onClick={this.changeName}>
+            <EditOutlined />
+          </EditButton>
+        </>
+      );
+    }
+  }
+
+  changeEmail(event) {
+    const id = +event.target.dataset.id;
+    const { changeEmail } = this.props;
+    changeEmail(id);
+  }
+
+  cancelEmailChanging(event) {
+    const id = +event.target.dataset.id;
+    const { cancelEmailChanging } = this.props;
+    cancelEmailChanging(id);
+  }
+
+  handleEmailChange(data) {
+    const { updateEmail, updatingId } = this.props;
+    data.id = updatingId;
+    updateEmail(data);
+  }
+
+  renderChangeEmailOrActualEmail(id, changeEmailIds) {
```

Comment:
Move it to the separated component. I mean switcher between text and input and generalize it to be used for names and emails

Reply:
Done